libs/blocks/global-navigation/utilities/keyboard/popup.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable class-methods-use-this */
+import {
+  getNextVisibleItem,
+  getPreviousVisibleItem,
+  getOpenPopup,
+  selectors,
+} from './utils.js';
+import { closeAllDropdowns } from '../utilities.js';
+
+const getState = ({ e } = {}) => {
+  const popupEl = getOpenPopup();
+  if (!popupEl) return {};
+  const popupItems = [...popupEl.querySelectorAll(selectors.popupItems)];
+  const curr = popupItems.findIndex((el) => el === e.target);
+  const prev = getPreviousVisibleItem(curr, popupItems);
+  const next = getNextVisibleItem(curr, popupItems);
+  const column = document.activeElement.closest(selectors.column);
+  const visibleColumns = [...popupEl.querySelectorAll(selectors.column)];
+  const currentColumn = visibleColumns.findIndex((node) => node.isEqualNode(column));
+  const prevColumn = visibleColumns[currentColumn - 1] || -1;
+  const nextColumn = visibleColumns[currentColumn + 1] || -1;
+
+  return {
+    popupItems,
+    curr,
+    prev,
+    next,
+    prevColumn,
+    nextColumn,
+  };
+};
+class Popup {
+  constructor({ mainNav }) {
+    this.mainNav = mainNav;
+    this.addEventListeners();
+    this.desktop = window.matchMedia('(min-width: 900px)');
+  }
+
+  open({ focus } = {}) {
+    const popupItems = [...(getOpenPopup()?.querySelectorAll(selectors.popupItems) || [])];
+    if (!popupItems.length) return;
+    const first = getNextVisibleItem(-1, popupItems);
+    const last = getPreviousVisibleItem(popupItems.length, popupItems);
+    if (focus === 'first') popupItems[first].focus();
+    if (focus === 'last') popupItems[last].focus();
+  }
+
+  addEventListeners = () => {
+    document.querySelector('header').addEventListener('keydown', (e) => {
```

Comment:
Same comment with `header` vs `.global-navigation`; and we'll also need to adapt once we have the footer.